Source/Managers/SceneMan.h

```diff
@@ -510,6 +510,23 @@ namespace RTE {
 		/// @return Whether any unseen pixels were revealed as a result of this seeing.
 		bool CastUnseenRay(int team, const Vector& start, const Vector& ray, Vector& endPos, int strengthLimit, int skip, bool reveal);
 
+		// TODO: Find a better name for this before release
+		/// Traces a box along a vector and reveals or hides pixels on the unseen layer of a team
+		/// as long as the accumulated material strengths traced through the terrain
+		/// don't exceed a specific value.
+		/// @param team The team to see for.
+		/// @param start The starting position.
+		/// @param ray The vector to trace along.
+		/// @param endPos A Vector that will be set to the position of where the sight ray was
+		/// terminated. If it reached the end, it will be set to the end of the ray.
+		/// @param size The minimum size of the box as it's cast.
+		/// @param strengthLimit The material strength limit where
+		/// @param skip For every pixel checked along the line, how many to skip between them
+		/// for optimization reasons. 0 = every pixel is checked.
+		/// @param reveal Whether the ray should reveal or restore unseen layer
+		/// @return Whether any unseen pixels were revealed as a result of this seeing.
+		bool CastUnseenBox(int team, const Vector& start, const Vector& ray, Vector& endPos, int size, int strengthLimit, int skip, bool reveal);
```

Comment:
Have you considered casting a circle instead of an AABB? Might give a prettier rounded revealed area at corners

Reply:
I did consider it back in the Lua tests, it actually looks really bad on flat walls. Circles have a really hard time filling in the depth of the wall, and show more gratingly that there are not a lot of casts. When I had it running, everyone who saw told me they liked how the corners looked before anyhow. A [Squircle](https://en.wikipedia.org/wiki/Squircle) might be a good middle ground, but it'd have to scale to any dimensions, so I don't know if that'd get costly performance wise.